### PR TITLE
lottie: replace uuid with custom unique id

### PR DESCRIPTION
**Pull request by Howl-Kim**
## Changes
Removes the `uuid` dependency across the project and replaces its usage with native alternatives or other utilities. Additionally, it cleans up related dependencies and configurations in various files.

### Replacement of `uuid` usage:

*  Updated `src/lottie-player.ts` to generate unique canvas IDs using a combination of `Date.now()` and `Math.random()` instead of `uuidv4`. [[1]](diffhunk://#diff-b00e87ab882980bad91534c8414d03d8137fbb1e3ecc49cf77a9daf4e38eb5f1L25) [[2]](diffhunk://#diff-b00e87ab882980bad91534c8414d03d8137fbb1e3ecc49cf77a9daf4e38eb5f1L421-R420)

### Removal of `uuid` dependency:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L50): Removed the `uuid` package and its type definitions (`@types/uuid`) from both `dependencies` and `devDependencies`. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L50) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L68-R67) [[3]](diffhunk://#diff-f084e1ad57f1e750dda5bd9177c78941e2b951b781a9fec039508d33fa3c2e2cL23-L30)
* [`rollup.config.js`](diffhunk://#diff-6814bf77564b4f1c92f5861e184e28fe217c080a047fefa8b73a728f755ec45cL14): Removed `uuid` from the `globals` configuration.

### CODE REVIEW

File: src/lottie-player.ts [hunk 1]
```diff
@@ -418,7 +417,7 @@ export class LottiePlayer extends LitElement {
   protected firstUpdated(_changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>): void {
     this._canvas = this.querySelector('.thorvg') as HTMLCanvasElement;
     
-    this._canvas.id = `thorvg-${uuidv4().replaceAll('-', '').substring(0, 6)}`;
+    this._canvas.id = `thorvg-${self.crypto.randomUUID().replaceAll('-', '').substring(0, 6)}`;
```
[the remark this answers is not in the document]
Comment (Howl-Kim) [reply src/lottie-player.ts:420]: If we need to handle this edge case, what is the proper alternative.

[the remark this answers is not in the document]
Comment (tinyjin) [reply src/lottie-player.ts:420]: @Howl-Kim `uuid` was the alternative. By this patch, users should ensure secure context in their environment. We should mull over whether this restriction is really edge case. What if users want to test in dev env or private network without HTTPS. This will be ungeneral case or general case? This is a question that we, as library providers, need to think deeply about.

**Comment by tinyjin** [comment]
@Howl-Kim It'd be better if we could find a way either [1] not to generate unique key fundamentally, or [2] own generation that ensures uniqueness

**Review by tinyjin — COMMENTED** [review]
Please remove change in `perf-test`. Removing uuid from `perf-test` isn't main goal. Just for `lottie-player` is pretty enough. Let's focus on atomic & meaningful change.

File: src/lottie-player.ts [hunk 2]
```diff
@@ -418,7 +417,7 @@ export class LottiePlayer extends LitElement {
   protected firstUpdated(_changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>): void {
     this._canvas = this.querySelector('.thorvg') as HTMLCanvasElement;
     
-    this._canvas.id = `thorvg-${uuidv4().replaceAll('-', '').substring(0, 6)}`;
+    this._canvas.id = `thorvg-${Date.now().toString(36) + Math.random().toString(36).substring(2, 9)}`;
```
[the remark this answers is not in the document]
Comment (tinyjin) [reply src/lottie-player.ts:420]: Also would be nicer if the line could be maintained in private helper function, something like this:

```ts
const _generateUID = () => {
  // generate uid
}
```

[the remark this answers is not in the document]
Comment (Howl-Kim) [reply src/lottie-player.ts]: @tinyjin Nice suggestion :) 0272a36dfd5954cc101cd0f4241e825a2d0043ac

**Comment by Howl-Kim** [comment]
@tinyjin squash completed :)

**Comment by tinyjin** [comment]
Good job @Howl-Kim
